src/core_ocean/analysis_members/mpas_ocn_lagrangian_particle_tracking.F

```diff
@@ -497,8 +521,13 @@ subroutine ocn_compute_lagrangian_particle_tracking(domain, timeLevel, err)!{{{
           call mpas_pool_get_array(particle % haloDataPool, 'currentCell', iCell)
 
           ! sampling variables
-          call mpas_pool_get_array(particle % haloDataPool, 'temperature', temperature)
-          call mpas_pool_get_array(particle % haloDataPool, 'salinity', salinity)
+          call mpas_pool_get_array(particle % haloDataPool, 'temperature', particleTemperature)
```

Comment:
@bradyrx, here is the bug causing the issue-- you changed the registry particle names from `salinity` to `particleSalinity` but didn't update that here.  I think this is why you have the seg fault... the get array command gives an error because there is no `temperature` field in `haloDataPool`, see above comment.  

I think this needs fixed unless I've missed something in your refractor.

Reply:
If indeed this is the case, and I suspect it is, then it is disappointing `DEBUG=true` did not give us enough information to catch this error because you'll note the debug message suggest the error was in the `interp_cell_scalars` routine, which wasn't changed at all from my tested set of commits and these commits.  Just something to keep in mind-- E3SM debug messages don't appear to be as rich as standalone ones.  This may be something worth pursuing more in the future and may even warrant an issue.